installer.sh.in

```diff
@@ -1091,14 +1091,51 @@ failed to create filesystem $fstype on $dev!\ncheck $LOG for errors." ${MSGBOXSI
 failed to mount $dev on ${mntpt}! check $LOG for errors." ${MSGBOXSIZE}
             DIE 1
         fi
+        # Create subvolumes on Btrfs systems
+        if [ "$fstype" = "btrfs" ]; then
+            btrfs subvolume create $TARGETDIR/@ >$LOG 2>&1
+            btrfs subvolume create $TARGETDIR/@cache >$LOG 2>&1
+            btrfs subvolume create $TARGETDIR/@log >$LOG 2>&1
+            # Create @home subvolume when wihtout /home partition
+            if ! echo "$mnts" | grep -q "/home"; then
+                btrfs subvolume create $TARGETDIR/@home >$LOG 2>&1
+            fi
+            umount $TARGETDIR
+            # Remount root, create mountpoints
+            mount -t $fstype -o subvol=@ $dev $TARGETDIR >$LOG 2>&1
+            mkdir -p $TARGETDIR/var/cache
+            mount -t $fstype -o subvol=@cache $dev $TARGETDIR/var/cache >$LOG 2>&1
+            mkdir $TARGETDIR/var/log
+            mount -t $fstype -o subvol=@log $dev $TARGETDIR/var/log >$LOG 2>&1
+            if ! echo "$mnts" | grep -q "/home"; then
+                mkdir $TARGETDIR/home
+                mount -t $fstype -o subvol=@home $dev $TARGETDIR/home >$LOG 2>&1
+            fi
+            if [ $? -ne 0 ]; then
+                DIALOG --msgbox "${BOLD}${RED}ERROR:${RESET} \
+failed to remount $dev on ${mntpt}! check $LOG for errors." ${MSGBOXSIZE}
+                DIE 1
+            fi
+        fi
         # Add entry to target fstab
         uuid=$(blkid -o value -s UUID "$dev")
         if [ "$fstype" = "f2fs" -o "$fstype" = "btrfs" -o "$fstype" = "xfs" ]; then
             fspassno=0
         else
             fspassno=1
         fi
-        echo "UUID=$uuid $mntpt $fstype defaults 0 $fspassno" >>$TARGET_FSTAB
+        mntopts=defaults
+        if [ "$fstype" = "btrfs" ]; then
+            mntopts="$mntopts,compress=zstd"
```

Comment:
Is enabling compression by default a good idea here?

Reply:
It can improve performance and increase SSD lifespan.